# Capstone Project - Museums Berlin

## Introduction / Business Problem

Many people **visit Berlin** for **cultural** reasons and explore the vast opportunities to visit **museums**. In order to **reduce the time** going to a museum or coming home, you want to **stay at a place** which located in the postal code area with the **highest density of museums**.

Therefor, in this project we will try to locate the areas in **Berlin** with the highest density of **museums**.

We will use our knowledge in data science to locate the area you want to book your hotel, AirBnB, etc. based on our criteria. The amount of museums within each postal area will then be clearly visualized in order to find the best possible location for your stay.

## Data

The data needed for our analysis can be retrieved via the **foursquare API** - Number of museums in each neighborhood / postal code area

## Methodology

In a first step we will retrieve all foursquare entries on museums in Berlin.

In a second step we will filter the data based on the postal codes of the locations in Berlin.

In a last step we will analyze the data according to the highest density postal code area with museums in Berlin.

## Analysis

In [13]:
#Import libraries
import requests
import pandas as pd 
import numpy as np

from geopy.geocoders import Nominatim # module to convert an address into latitude and longitude values

import folium # plotting library

from pandas.io.json import json_normalize

In [9]:
#Define foursquare credentials
CLIENT_ID = '5GU4TM5OKKJNTW01ANMGESFQ2QI3K2AKQWCS3WKPU3BCRDJT' # your Foursquare ID
CLIENT_SECRET = 'BEM55QHRBTDSNZYMRHKUUX21CBZH5BODNYS4FVQDTOU4RJX0' # your Foursquare Secret
ACCESS_TOKEN = 'L30HL4MSA1A1EDYY2EFC3IU3UPU1SPZ3NDNXR2YAZQNNH23S' # your FourSquare Access Token
VERSION = '20180605' # Foursquare API version
LIMIT = 100 # A default Foursquare API limit value

In [6]:
#Define location
address = 'Berlin'

geolocator = Nominatim(user_agent="foursquare_agent")
location = geolocator.geocode(address)
latitude = location.latitude
longitude = location.longitude
print(latitude, longitude)

52.5170365 13.3888599


In [18]:
#Define search query
search_query = 'Museum'
radius = 2000


In [19]:
#Define corresponding url
url = 'https://api.foursquare.com/v2/venues/search?client_id={}&client_secret={}&ll={},{}&oauth_token={}&v={}&query={}&radius={}&limit={}'.format(CLIENT_ID, CLIENT_SECRET, latitude, longitude,ACCESS_TOKEN, VERSION, search_query, radius, LIMIT)
url

'https://api.foursquare.com/v2/venues/search?client_id=5GU4TM5OKKJNTW01ANMGESFQ2QI3K2AKQWCS3WKPU3BCRDJT&client_secret=BEM55QHRBTDSNZYMRHKUUX21CBZH5BODNYS4FVQDTOU4RJX0&ll=52.5170365,13.3888599&oauth_token=L30HL4MSA1A1EDYY2EFC3IU3UPU1SPZ3NDNXR2YAZQNNH23S&v=20180605&query=Museum&radius=2000&limit=100'

In [81]:
#Examine results
results = requests.get(url).json()

# assign relevant part of JSON to venues
venues = results['response']['venues']


# tranform venues into a dataframe
dataframe = json_normalize(venues)
dataframe['location.postalCode'] = dataframe['location.postalCode'].fillna(0)
dataframe.rename(columns={'location.lat': 'lat', 'location.lng': 'lng'}, inplace = True)

dataframe


<ipython-input-81-1cecb3d0ae2d>:9: FutureWarning: pandas.io.json.json_normalize is deprecated, use pandas.json_normalize instead
  dataframe = json_normalize(venues)


id  \
0   4ed2360702d5feaa1e6594e6   
1   4adcda83f964a520aa4821e3   
2   4adcda83f964a520b24821e3   
3   4adcda81f964a5207d4821e3   
4   4adcda81f964a5205d4821e3   
5   4adcda81f964a520574821e3   
6   4adcda81f964a5205b4821e3   
7   4adcda80f964a5201c4821e3   
8   4ae9c897f964a52062b621e3   
9   4b13cbcbf964a5203d9923e3   
10  4adcda81f964a520594821e3   
11  50ac9beee4b0a95638d048d0   
12  4adcda81f964a5205c4821e3   
13  508bce4ee4b0c56e752e443d   
14  50f6c465e4b052e422df4aa8   
15  4c4d697f9efabe9a80aa7368   
16  4adcda81f964a520584821e3   
17  4adcda83f964a520ab4821e3   
18  54e380ef498e9832a2b37885   
19  4c6fb1f4fa49a1cd4eb3a3e3   
20  52725782498e92d2fe321639   
21  5480b06b498efa1757baa290   
22  4eca77e5f7904135202d36af   
23  4f42de33e4b0a2bdc6c470ce   
24  519a2d62498ebfecbf8e51c6   
25  4b4cd624f964a520f4c026e3   
26  4adcda81f964a5207e4821e3   
27  4bf2ce35e5eba5930fd21d90   
28  4f9fba6ee4b0a2e4c2f89e34   
29  4adcda81f964a5207c4821e3   
30  4adcda81f964a520564821e3   
31  4e171fc18877d38dc90c2363   
32  4bd185f841b9ef3b7b05fce5   
33  4d6cd738125eb60c2e322c4e   
34  5ad0d424a879213b8771957e   
35  5b758f0a2be425002b1b8c68   
36  59eefb25d3cce84392c1ef41   
37  4e0768777d8b2a01805a661b   
38  544e3167498e9587ae530935   
39  573715fd498e06f096843c49   
40  4d2de9206fc6f04d30027d45   
41  4adcda81f964a5205f4821e3   
42  4ce90079baa6a1cd5a6f316c   
43  526f8c3f498eb4538f017f25   
44  59c3dd59898bdc23f9190357   
45  5f5f288fa154fa03f70f2bc7   
46  55410ee1498e208aca736d81   
47  57e15eff498e4cd0ba163948   
48  596f8fbf6f706a4c51cdb3ac   
49  4ae44d24f964a520eb9921e3   

                                                 name  \
0                 Antik- und Buchmarkt am Bode-Museum   
1   German Historical Museum (Deutsches Historisch...   
2                                        Neues Museum   
3                            Museum für Kommunikation   
4                         Museum für Islamische Kunst   
5                                        Altes Museum   
6                                         Bode-Museum   
7                        Museum Island (Museumsinsel)   
8                                          DDR Museum   
9                                 Café im Bode-Museum   
10                              Museum für Naturkunde   
11  Bibliothek / Bildarchiv Deutsches Historisches...   
12  Skulpturensammlung und Museum für Byzantinisch...   
13                       Münzkabinett im Alten Museum   
14                               Café im Alten Museum   
15          Mauer Museum - Haus am Checkpoint Charlie   
16     Jewish Museum Berlin (Jüdisches Museum Berlin)   
17           Hamburger Bahnhof – Museum für Gegenwart   
18                               MeMu Menschen Museum   
19  St. Nicholas' Church Museum (Museum Nikolaikir...   
20                                       Trabi Museum   
21                        MAGICUM Berlin Magic Museum   
22                H U Märkisches Museum / Inselstraße   
23  Designpanoptikum - surreales Museum für indust...   
24                                       Zille Museum   
25                                U Märkisches Museum   
26                                  Märkisches Museum   
27  Deutsche Kinemathek - Museum für Film und Fern...   
28                                 H Jüdisches Museum   
29                           Musikinstrumenten Museum   
30                                        Hanf Museum   
31                                DDR Motorrad-Museum   
32             Medizinhistorisches Museum der Charité   
33                           Historiale Berlin Museum   
34                  Museum Frieder Burda Salon Berlin   
35                              Museum der Illusionen   
36                           Brandenburger Tor Museum   
37                    Sonntagskonzerte am Bode-Museum   
38             Ägyptisches Museum und Papyrussammlung   
39                           Vorderasiatisches Museum   
40                 Museum Blindenwerkstatt Otto Weidt   

In [65]:
dataframe.shape

(50, 19)

In [67]:
#Drop rows with no postal codes
indexNames = dataframe[dataframe['location.postalCode'] == 0].index
dataframe.drop(indexNames , inplace=True)

dataframe.shape

(43, 19)

In [87]:
#Visualize venues (museums) on map

venues_map = folium.Map(location=[latitude, longitude], zoom_start=100)

for lat, lng, label in zip(dataframe['lat'], dataframe['lng'], dataframe['name']):
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        color='blue',
        popup=label,
        fill = True,
        fill_color='blue',
        fill_opacity=0.6
    ).add_to(venues_map)

# display map
venues_map

In [69]:
#Count postal codes
postalcode_counts = dataframe['location.postalCode'].value_counts()
postalcode_counts

10178    18
10117     9
10785     5
10179     4
10969     4
10115     2
10557     1
Name: location.postalCode, dtype: int64

## Results and Discussion

The **postal code area 10178** has **18 museums**, which is double the amount of the second highest density of museums in specific postal code areas of Berlin.
The assumption would be that there is a dense cluster of museums in that area. It **might be a bigger complex with several museums** in it. 
The **type of museum is not display** so it would need **further analysis** regarding the types of museums.

## Conclusion

Based on the data it would be **recommended for a cultural visit to Berlin** to search your **accommodation within** the postal code area of **10178**.